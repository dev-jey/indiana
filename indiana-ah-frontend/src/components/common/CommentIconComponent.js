import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import CommentIcon from '../../assets/images/icons/chat-1.png';

const CommentIconComponent = ({ commentCount }) => (
<Fragment>
<span className = 'icon-holder'>
<img src = {CommentIcon} alt = 'CommentIcon' className= 'icon-color icon-height icon_margin'/>
<sub className='count'>{commentCount || ''}</sub>
</span>
</Fragment>
);

CommentIconComponent.propTypes = {
  commentCount: PropTypes.number.isRequired,
};

export default CommentIconComponent;
