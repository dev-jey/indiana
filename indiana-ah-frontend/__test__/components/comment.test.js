import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CommentIconComponent from '../../src/components/common/CommentIconComponent';

describe('<CommentIconComponent />', () => {
  const wrapper = shallow(<CommentIconComponent commentCount={5} />);
  it('should render a comment component', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper.find('span')).toHaveLength(1);
    expect(wrapper.find('img')).toHaveLength(1);
    expect(wrapper.find('sub')).toHaveLength(1);
  });
  const commentwrapper = shallow(<CommentIconComponent commentCount={0} />);
  it('should not render a comment count if it is zero', () => {
    expect(commentwrapper.find('sub').text()).toEqual('');
  });
  it('renders correctly', () => {
    const tree = renderer.create(<CommentIconComponent commentCount={5} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
