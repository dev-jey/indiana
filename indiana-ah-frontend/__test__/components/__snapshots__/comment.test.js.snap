// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentIconComponent /> renders correctly 1`] = `
<span
  className="icon-holder"
>
  <img
    alt="CommentIcon"
    className="icon-color icon-height icon_margin"
    src="test-file-stub"
  />
  <sub
    className="count"
  >
    5
  </sub>
</span>
`;
